bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/DevModeOperations.java

```diff
@@ -763,6 +779,179 @@ public void startDevMode(String cmd, String projectName, String projectPath, Str
+                            if (monitor.isCanceled()) {
+                                p.destroy();
+                                return Status.CANCEL_STATUS;
+                            }
+                            completed = p.waitFor(1, TimeUnit.SECONDS);
+                        }
+
+                        if (!completed) {
+                            setProperty(STOP_JOB_COMPLETION_TIMEOUT, Boolean.TRUE);
+                        } else {
+                            setProperty(STOP_JOB_COMPLETION_EXIT_CODE, p.exitValue());
+                            if (p.exitValue() != 0) {
+                                BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+                                StringBuilder builder = new StringBuilder();
+                                String line = null;
+                                while ((line = reader.readLine()) != null) {
+                                    builder.append(line);
+                                    builder.append(System.getProperty("line.separator"));
+                                }
+                                setProperty(STOP_JOB_COMPLETION_OUTPUT, builder.toString());
+                            }
+                        }
+                    } catch (Exception e) {
+                        ErrorHandler.processErrorMessage("Exception issuing plugin stop command", e, false);
+                    }
+                    return Status.OK_STATUS;
+                }
+
+            };
+
+            job.addJobChangeListener(new JobChangeAdapter() {
+                @Override
+                public void done(IJobChangeEvent event) {
+
+                    runningJobs.remove(event.getJob());
+
+                    /*
+                     * Check for timeout
+                     */
+                    Object completion = event.getJob().getProperty(STOP_JOB_COMPLETION_TIMEOUT);
+                    if (Boolean.TRUE.equals(completion)) {
+                        // Need to do this on main thread since it's displayed to the user.
+                        Display.getDefault().syncExec(new Runnable() {
+                            @Override
+                            public void run() {
+
+                                String msg = "The Liberty Maven or Gradle stop command issued for project " + projectName
+                                        + " timed out after " + STOP_TIMEOUT_SECONDS + " seconds.";
+                                if (Trace.isEnabled()) {
+                                    Trace.getTracer().trace(Trace.TRACE_TOOLS, msg);
+                                }
+                                ErrorHandler.rawErrorMessageDialog(msg);
+                            }
+                        });
+                    }
+
+                    /*
+                     * Check for bad exit value
+                     */
+                    Object rc = event.getJob().getProperty(STOP_JOB_COMPLETION_EXIT_CODE);
```

Comment:
The rc value can be null if the job is cancelled before it completes. Line 867 will exit the job and one would get a message with a null exit value: "Stop failed with exitValue = null ...". Also the message can be a bit misleading in this case (stop may have worked because of timing), so perhaps along with checking the the rc value for null, etc, perhaps no message would be better unless we detected that the cancel failed?